test/fuzz/Rebalancer.t.sol

```diff
@@ -0,0 +1,636 @@
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function test_RebalancerInitialization() public {
+        assertEq(address(rebalancer.accessControlManager()), address(accessControlManager));
+        assertEq(address(rebalancer.agToken()), address(agToken));
+        assertEq(address(rebalancer.transmuter()), address(transmuter));
+    }
+
+    function test_Constructor_RevertWhen_ZeroAddress() public {
+        vm.expectRevert(Errors.ZeroAddress.selector);
+        new Rebalancer(IAccessControlManager(address(0)), transmuter);
+
+        vm.expectRevert();
+        new Rebalancer(accessControlManager, ITransmuter(address(0)));
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                       SET ORDER                                                    
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function test_SetOrder_RevertWhen_NonGovernorOrGuardian() public {
+        vm.expectRevert(Errors.NotGovernorOrGuardian.selector);
+        rebalancer.setOrder(address(eurA), address(eurB), 100, 1);
+    }
+
+    function test_SetOrder_RevertWhen_InvalidParam() public {
+        vm.startPrank(governor);
+        vm.expectRevert(Errors.InvalidParam.selector);
+        rebalancer.setOrder(address(eurA), address(eurB), 100, 1);
+        vm.stopPrank();
+    }
+
+    function testFuzz_SetOrder(
+        uint256 subsidyBudget,
+        uint256 guaranteedRate,
+        uint256 subsidyBudget1,
+        uint256 guaranteedRate1
+    ) public {
+        uint256 a;
+        uint256 b;
+        subsidyBudget = bound(subsidyBudget, 10 ** 9, 10 ** 27);
+        guaranteedRate = bound(guaranteedRate, 10 ** 15, 10 ** 21);
+        subsidyBudget1 = bound(subsidyBudget1, 10 ** 9, 10 ** 27);
+        guaranteedRate1 = bound(guaranteedRate1, 10 ** 15, 10 ** 21);
+        deal(address(agToken), address(rebalancer), subsidyBudget);
+        vm.startPrank(governor);
+        rebalancer.setOrder(address(eurA), address(eurB), subsidyBudget, guaranteedRate);
+        vm.stopPrank();
+        (a, b) = rebalancer.orders(address(eurA), address(eurB));
+        assertEq(a, subsidyBudget);
+        assertEq(b, guaranteedRate);
+        assertEq(rebalancer.budget(), subsidyBudget);
+        assertEq(
+            rebalancer.getGuaranteedAmountOut(address(eurA), address(eurB), decimalsA),
+            (guaranteedRate * decimalsB) / 1e18
+        );
+
+        vm.startPrank(governor);
+        vm.expectRevert(Errors.InvalidParam.selector);
+        rebalancer.setOrder(address(eurA), address(eurB), subsidyBudget + 1, guaranteedRate + 1);
+        vm.stopPrank();
+
+        vm.startPrank(governor);
+        rebalancer.setOrder(address(eurA), address(eurB), subsidyBudget / 3, guaranteedRate);
```

Comment:
can fuzz the `/3` and `+1`

Reply:
Yes I have the fuzzing test above, the goal of this test was more to make sure the logic works